lib/internal/modules/esm/get_format.js

```diff
@@ -71,9 +69,19 @@ function defaultGetFormat(url, context, defaultGetFormatUnused) {
         throw new ERR_UNKNOWN_FILE_EXTENSION(ext, fileURLToPath(url));
       }
     }
-    return { format: format || null };
-  }
-  return { format: null };
+
+    return format || null;
+  },
+  'node:'() { return 'builtin'; },
+  noop() { return null; },
+};
+
+function defaultGetFormat(url, context) {
+  const parsed = new URL(url);
+
+  return (
+    protocolHandlers[parsed.protocol] ?? protocolHandlers.noop
+  )(parsed, url);
```

Comment:
`protocolHandlers` inherits from `Object.prototype`, we need to account prototype pollution:
```suggestion
  return ObjectPrototypeHasOwnProperty(protocolHandlers, parsed.protocol) ?
    protocolHandlers[parsed.protocol](parsed, url) :
    null;
```

Reply:
I think I would rather create a null prototype object then for `protocolHandlers` (smaller object = less memory, plus prototype pollution protection). Thoughts?

Reply:
Using null prototype object doesn't change anything memory-wise IIRC. Also my suggestion avoids a function call if the protocol is unknown, which is always nice. That is to say: I wouldn't go for a null prototype if I were the one doing this PR, but I don't object to it.